src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/InstalledAndTransitivePackageFeed.cs

```diff
@@ -25,59 +25,51 @@ public InstalledAndTransitivePackageFeed(IEnumerable<PackageCollectionItem> inst
+            IPackageSearchMetadata[] searchItems = intalledItems.Concat(transitiveItems).ToArray();
 
-            IEnumerable<IPackageSearchMetadata> items = await TaskCombinators.ThrottledAsync(
-                allPkgs,
-                (p, t) => GetPackageMetadataAsync(p, searchToken.SearchFilter.IncludePrerelease, t),
-                cancellationToken);
-
-            return CreateResult(items);
+            return CreateResult(searchItems);
         }
 
-        internal async Task<IPackageSearchMetadata> GetPackageMetadataAsync(PackageCollectionItem identity, bool includePrerelease, CancellationToken cancellationToken)
+        internal override async Task<IPackageSearchMetadata> GetPackageMetadataAsync<T>(T pkgIdentity, bool includePrerelease, CancellationToken cancellationToken)
         {
-            IEnumerable<ITransitivePackageReferenceContextInfo> transitivePRs = identity.PackageReferences.OfType<ITransitivePackageReferenceContextInfo>();
-
-            ITransitivePackageReferenceContextInfo transitivePR = transitivePRs.OrderByDescending(x => x.Identity.Version).FirstOrDefault();
+            cancellationToken.ThrowIfCancellationRequested();
 
-            if (transitivePR != null)
+            if (pkgIdentity is PackageCollectionItem identity)
             {
-                IReadOnlyCollection<PackageIdentity> transitiveOrigins = transitivePR.TransitiveOrigins?.Select(to => to.Identity).ToArray() ?? Array.Empty<PackageIdentity>();
+                IEnumerable<ITransitivePackageReferenceContextInfo> transitivePRs = identity.PackageReferences.OfType<ITransitivePackageReferenceContextInfo>();
 
-                if (transitiveOrigins.Any())
+                ITransitivePackageReferenceContextInfo transitivePR = transitivePRs.OrderByDescending(x => x.Identity.Version).FirstOrDefault();
+
+                if (transitivePR != null)
                 {
-                    // Get only local metadata. We don't want Deprecation and Vulnerabilities Metadata on Transitive packages
-                    IPackageSearchMetadata packageMetadata = await _metadataProvider.GetOnlyLocalPackageMetadataAsync(identity, cancellationToken);
+                    IReadOnlyCollection<PackageIdentity> transitiveOrigins = transitivePR.TransitiveOrigins?.Select(to => to.Identity).ToArray() ?? Array.Empty<PackageIdentity>();
 
-                    if (packageMetadata == null) // Edge case: local metadata not found
+                    if (transitiveOrigins.Any())
                     {
-                        packageMetadata = PackageSearchMetadataBuilder.FromIdentity(identity).Build(); // create metadata object only with ID
-                    }
+                        // Get only local metadata. We don't want Deprecation and Vulnerabilities Metadata on Transitive packages
+                        IPackageSearchMetadata packageMetadata = await _metadataProvider.GetOnlyLocalPackageMetadataAsync(identity, cancellationToken);
```

Comment:
extra whitespace ?

Reply:
Extra whitespace (indentation) is needed because I added a new condition to check if param `pkgIdentity` (now renamed back to identity) if of type PackageCollectionItem

Reply:
Ok - here's how GitHub shows it to me.. wasn't sure if accurate:
![image](https://user-images.githubusercontent.com/49205731/157997535-d0192e62-128d-4a74-a5ed-542b8e55e83a.png)